ospackage/recipes/buildcommand.go

```diff
@@ -46,10 +48,64 @@ func BuildCommand(step *osconfigpb.SoftwareRecipe_Step, artifacts map[string]str
 
 // StepFileCopy builds the command for a FileCopy step
 func StepFileCopy(step *osconfigpb.SoftwareRecipe_Step_FileCopy, artifacts map[string]string) ([]string, error) {
-	fmt.Println("StepFileCopy")
+	dest, err := normPath(step.FileCopy.Destination)
+	if err != nil {
+		return nil, err
+	}
+
+	permissions, err := parsePermissions(step.FileCopy.Permissions)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := os.Stat(dest); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	} else {
+		// file exists
+		if !step.FileCopy.Overwrite {
+			return nil, fmt.Errorf("in FileCopy step file already exists at path %q and Overwrite = false", step.FileCopy.Destination)
+		}
+		os.Chmod(dest, permissions)
+	}
+
+	src, ok := artifacts[step.FileCopy.ArtifactId]
+	if !ok {
+		return nil, fmt.Errorf("Could not find location for artifact %q in FileCopy step", step.FileCopy.ArtifactId)
```

Comment:
The previous error starts with `in FileCopy step`, should this one also start with that or is there a reason to be inconsistent?

Reply:
I took out "in FileCopy step" and changed installrecipe to wrap the error with the step number to make debugging easier